core/templates/pages/collection-player-page/collection-player-page.component.ts

```diff
@@ -0,0 +1,433 @@
+    //   }
+    // }, true);
+    this.loaderService.showLoadingScreen('Loading');
+    this.collection = null;
+    this.collectionId = this.urlService.getCollectionIdFromUrl();
+    this.explorationCardIsShown = false;
+    // The pathIconParameters is an array containing the co-ordinates,
+    // background color and icon url for the icons generated on the
+    // path.
+    this.pathIconParameters = [];
+    this.activeHighlightedIconIndex = -1;
+    this.MIN_HEIGHT_FOR_PATH_SVG_PX = 220;
+    this.ODD_SVG_HEIGHT_OFFSET_PX = 150;
+    this.EVEN_SVG_HEIGHT_OFFSET_PX = 280;
+    this.ICON_Y_INITIAL_PX = 35;
+    this.ICON_Y_INCREMENT_PX = 110;
+    this.ICON_X_MIDDLE_PX = 225;
+    this.ICON_X_LEFT_PX = 55;
+    this.ICON_X_RIGHT_PX = 395;
+    this.svgHeight = this.MIN_HEIGHT_FOR_PATH_SVG_PX;
+    this.nextExplorationId = null;
+    this.whitelistedCollectionIdsForGuestProgress = (
+      AppConstants.WHITELISTED_COLLECTION_IDS_FOR_SAVING_GUEST_PROGRESS);
+    // $anchorScroll.yOffset = -80;
+
+    // Touching anywhere outside the mobile preview should hide it.
+    document.addEventListener('touchstart', () => {
+      if (this.explorationCardIsShown === true) {
+        this.explorationCardIsShown = false;
+      }
+    });
+
+    // This.collectionPlayerBackendApiService.someFunction(this.collectionId);
+
+    // this.collectionPlayerBackendApiService.fetchCollectionSummariesAsync(
+    //   this.collectionId);
+    this.http.get(
+      '/collection_handler/data/' + this.collectionId).toPromise().then(
+      function(response: CollectionHandler) {
+        angular.element('meta[itemprop="name"]').attr(
+          'content', response.meta_name);
+        angular.element('meta[itemprop="description"]').attr(
+          'content', response.meta_description);
+        angular.element('meta[property="og:title"]').attr(
+          'content', response.meta_name);
+        angular.element('meta[property="og:description"]').attr(
+          'content', response.meta_description);
+      }
+    );
+
+    // This.collectionPlayerBackendApiService.fetchCollectionSummaries().then(
+    //   (response: CollectionSummary) => {
+    //     this.collectionSummary = response.summaries[0];
+    //   }, () => {
+    //     this.alertsService.addWarning(
+    //       'There was an error while fetching the collection summary.');
+    //   }
+    // );
+    this.http.get('/collectionsummarieshandler/data', {
+      params: {
+        stringified_collection_ids: JSON.stringify([this.collectionId])
+      }
+    }).toPromise().then(
+      (response: CollectionSummary) => {
+      },
+      () => {
+        this.alertsService.addWarning(
+          'There was an error while fetching the collection summary.');
+      }
+    );
+    // Load the collection the learner wants to view.
+    this.readOnlyCollectionBackendApiService.loadCollectionAsync(
+      this.collectionId).then(
+      (collection) => {
+        this.collection = collection;
```

Comment:
change this to a function call, which assigns the new value to `this.collection` (name it maybe, `updateCollection`) and also executes the code in the callback which was used with `$scope.$watch`

Reply:
This correctly  fits in after usage of `ngOnChanges`